### PR TITLE
improve local development experience

### PR DESCRIPTION
**Pull request by mydiemho**
1. add api test file for use with REST Client extension
1. Ignore local appsettings file to prevent accidentally commiting
   credentials

### CODE REVIEW

**Comment by mydiemho** [comment]
> To make sure that my understanding is correct:
> 
> * New Dev will create the appsettings.Development.json
> * Use appsettings.sample.json as an example to populate appsettings.Development.json
> * Set ASPNETCORE_ENVIRONMENT=Development

Yup, didn't want to modify README and duplicate your work so will let you made the addition :)

ps: would be great to also explicitly mentioned that they will have to edit the cosmos connection string & auth key, plus instruction on where to retrieve that information

**Comment by michaeldeongreen** [comment]
> @michaeldeongreen
> Yup, didn't want to modify README and duplicate your work so will let you made the addition :)
> 
> ps: would be great to also explicitly mentioned that they will have to edit the cosmos connection string & auth key, plus instruction on where to retrieve that information

Yep, already have that in the README.  Will add what I wrote as well, just wanted to make sure I understood.

File: api/validate-api.http [hunk 1]
```diff
@@ -0,0 +1,46 @@
+### Save Bundle
```
Comment (alkofahi) [inline api/validate-api.http]: is there a way for us to automate this as a post-deployment validation via pipelines?

Comment (alkofahi) [reply api/validate-api.http]: actually, if we keep it this way, can you add a little comment describing how to utilize this :)

Comment (michaeldeongreen) [reply api/validate-api.http]: I plan to add a link or verbiage in the README on how to use the tool.

File: api/validate-api.http [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+### Get Bundle for Vanity URL
+GET https://localhost:5001/api/links/postman-test
+
+
+### Validate Page
+POST https://localhost:5001/api/validatePage
+Content-Type: application/json
+
+{
+	"url" : "marketplace.visualstudio.com/items? itemName=sdras vue-vscode-extensionpack",
```
Comment (Humoiz) [inline api/validate-api.http]: Have you tested if this url value returns a response?

File: api/validate-api.http [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      "id": "cjsorrc5d00013h5p3e1f8tgk",
+      "url": "google.com",
+      "title": "Google",
+      "description": "Search the world's information, including webpages, images, videos and more. Google has many special features to help you find exactly what you're looking for.",
+      "image": ""
+    }
+  ],
+  "vanityUrl": "postman-test",
+  "description": "",
+  "userId": "cecilphillip"
```
Comment (Humoiz) [inline api/validate-api.http]: Can you remove "userId" property entirely or provide empty value. The value for userId is generated in the code and not set from the body.

File: api/validate-api.http (hunk 2 above)
Comment (Humoiz) [reply api/validate-api.http]: I checked that it returns [
    {
        "id": "1",
        "title": null,
        "description": null,
        "image": null
    }
]

Can we change the url value to something else? May be google.com?

File: api/src/LinkyLink/appsettings.sample.json [hunk 4]
```diff
@@ -0,0 +1,17 @@
+{
```
Comment (wbreza) [inline api/src/LinkyLink/appsettings.sample.json]: Why add this into a `sample` extension vs just including it in the top level `appsettings.json` ?

File: api/validate-api.http (hunk 1 above)
Comment (wbreza) [reply api/validate-api.http]: This is more for dev time checking.  We would need to implement some of these as integration tests for post build integration checks.

**Comment by mydiemho** [comment]
> I think we should not delete appsettings.json file otherwise LGTM.
> 
> One thing to check before you merge this PR. Deploy this app to App Service and see if it works without the appsettings.json file. We are using DatabaseName & Container name from appsettings.json. Otherwise I can set these attributes as part of the deployment.

@Humoiz it's not deleted, just rename so user are aware it's the base setting

File: api/src/LinkyLink/appsettings.sample.json (hunk 4 above)
Comment (mydiemho) [reply api/src/LinkyLink/appsettings.sample.json]: I feel like naming it `sample` makes it clearer what kind of settings one can use with the app.  It just a practice that some language use.  No strong feelings either way and can change it back

File: api/validate-api.http (hunk 1 above)
Comment (mydiemho) [reply api/validate-api.http]: Yup, this is mainly for new developers to validate that they have the app set up correctly

File: api/validate-api.http (hunk 2 above)
Comment (mydiemho) [reply api/validate-api.http]: sure thing

File: api/validate-api.http (hunk 3 above)
Comment (mydiemho) [reply api/validate-api.http]: wasn't aware,  this was just rewritten base on the existing postman collection.  Will update this and the postman collection

Comment (mydiemho) [reply api/validate-api.http]: so I checked it again, and userId is empty on get (after a create).  The code might not be working as you expect

Comment (Humoiz) [reply api/validate-api.http]: It will be empty if the user is not authentication from Twitter.

Comment (mydiemho) [reply api/validate-api.http]: in that case, this is for testing local backend, hence won't have any Twitter.  I am incline to keep the userId with some dummy value

**Comment by mydiemho** [comment]
> I think we should not delete appsettings.json file otherwise LGTM.
> 
> One thing to check before you merge this PR. Deploy this app to App Service and see if it works without the appsettings.json file. We are using DatabaseName & Container name from appsettings.json. Otherwise I can set these attributes as part of the deployment.

That seems strange to me that we're only using the DatabaseName & ContainerName from `appsettings.json` but not the cosmosDB connection string

**Comment by Humoiz** [comment]
> > I think we should not delete appsettings.json file otherwise LGTM.
> > One thing to check before you merge this PR. Deploy this app to App Service and see if it works without the appsettings.json file. We are using DatabaseName & Container name from appsettings.json. Otherwise I can set these attributes as part of the deployment.
> 
> That seems strange to me that we're only using the DatabaseName & ContainerName from `appsettings.json` but not the cosmosDB connection string

I dont want to hardcode the connection string into appsettings.json. Anyway at this point I have setup connection string, database name & container name as part of the CD pipeline so no dependency here for production.

**Comment by mydiemho** [comment]
> > > I think we should not delete appsettings.json file otherwise LGTM.
> > > One thing to check before you merge this PR. Deploy this app to App Service and see if it works without the appsettings.json file. We are using DatabaseName & Container name from appsettings.json. Otherwise I can set these attributes as part of the deployment.
> > 
> > 
> > That seems strange to me that we're only using the DatabaseName & ContainerName from `appsettings.json` but not the cosmosDB connection string
> 
> I dont want to hardcode the connection string into appsettings.json. Anyway at this point I have setup connection string, database name & container name as part of the CD pipeline so no dependency here for production.

base on this [guide](https://docs.microsoft.com/en-us/archive/blogs/waws/asp-net-core-settings-for-azure-app-service), `appsettings.json` shouldn't be check in at all, which I agree with.

**Pull request merged**